rust-src/src/lib.rs

```diff
@@ -34,12 +34,25 @@ macro_rules! c_str {
 #[macro_export]
 macro_rules! transform_c_args {
     ($var:ident, $argc:expr, $argv:expr) => {
-        // TODO: `OsStr::from_bytes` only exists on *nix
-        use ::std::os::unix::ffi::OsStrExt;
         let $var: Vec<_> = (0..$argc)
-        .map(|i| unsafe { ::std::ffi::CStr::from_ptr(*$argv.add(i as usize)) })
-        .map(|i| ::std::ffi::OsStr::from_bytes(i.to_bytes()))
-        .collect();
+            .map(|i| unsafe { ::std::ffi::CStr::from_ptr(*$argv.add(i as usize)) })
+            .map(|x| {
+                ::cfg_if::cfg_if! {
+                    if #[cfg(unix)] {
+                        use ::std::os::unix::ffi::OsStrExt;
+                        ::std::ffi::OsStr::from_bytes(x.to_bytes())
+                    } else {
+                        // If arbitrary-bytes `OsStr`s are not supported (e.g. MS Windows),
```

Comment:
No, this is crap. We're not supporting non-unix targets any time soon, and if we do we definitely won't be dumping #ifdefs in the code like this - I'll expect this to be written with a proper helper.

Reply:
Alright and yes, helper "Ext" traits can be used here. And due to non-unix targets are not considered in a short time, for now I'm just going to revert this.